### PR TITLE
Color-String Makeover

### PR DESCRIPTION
**Pull request by Qix-**
Alright, the next step in the eventual [`color`](/Qix-/color) makeover.

The reason behind this PR is to allow better enumeration of methods in order to allow automatic conversion within the `color` module, without having to piece together strings or manually specify conversion functions. It'll keep the eventual `color` makeover even cleaner.

---

This PR makes slightly less drastic changes than the [`color-convert` makeover](Qix-/color-convert#27), but still clean things up a little bit. This will be released as a major version.

There are two commits to this PR; in the event the second PR is too magic for peoples' tastes, it'll be reverted.

Commit 1:

- `.getXXX()` functions -> `.get.xxx()`
- `.xxxString()` functions -> `.to.xxx()`
- `.getXXXa()` functions removed; all `.getXXX()` functions always return alpha
- `.percent[a]String()` -> `.to.rgb.percent()`
- Swizzling! See below.
- removed the `.getAlpha()` function - just use `.get.xxx(...)[3]`. This is partially because there is no real analogous `.to.alpha()` function.
- Under the Hood: `scale()` -> `clamp()`
- Under the Hood: `color-string.js` -> `index.js`

Commit 2:

- `.xxxaString()` functions (note the `a`, for alpha-enabled strings) -> `.to.xxx()`; the returned string is determined if alpha has been supplied to the call and if it is not `1`. In such a case, `rgb(...)` and `hsl(...)` are returned. In the event alpha was supplied and is a value other than `1`, then `rgba(...)` and `hsla(...)` are used.

---

Swizzling has been added. While a slight misnomer, this provides the ability to construct function parameters by passing a mixture of values and arrays.

```js
var cs = require('color-string');

cs.to.rgb(143, [16, 123], 0.5); // rgba(143, 16, 123, 0.5)
```

This is what will keep compatibility with the previous `.rgbaString(color, alpha)` calls, while adding the ability to accommodate code that does more per-channel stuff. There is little overhead in this change, other than an added dependency ([`simple-swizzle`](/Qix-/node-simple-swizzle)).

This change also cleans up the code quite a bit.

---

In the event the second commit is deemed "too magical" we can definitely remove it. However, with both commits in place, the API has been reduced to this:

```js
.get.rgb()
.get.hsl()
.get.hwb()

.to.rgb()
.to.rgb.percent()
.to.hsl()
.to.hwb()
.to.hex()
.to.keyword()
```

---

Would love some feedback, especially regarding the second commit.

// @sindresorhus @MoOx @kevinSuttle @OEvgeny @jbnicolai

### CODE REVIEW

**Comment by MoOx** [comment]
:+1:  * :100:

**Comment by Qix-** [comment]
Also apologies for moving the file around; Github doesn't understand files being moved and modified at the same time :|

**Comment by MoOx** [comment]
git should be able to do so, unless you changed too many (which is the case) and the diff consider this as a new file (which is kind of the case too right?). Or maybe it's just github yeah.

**Comment by Qix-** [comment]
Ah nope turns out Git didn't catch it as a rename.

```
deleted:    color-string.js
new file:   index.js
```

Shucks.

**Comment by kevinSuttle** [comment]
:+1:

**Comment by MoOx** [comment]
@Qix- once I even tried to use git mv, then changed it a lot: detected as a delete/new file :(

**Comment by Qix-** [comment]
Yeah there isn't a great way to approach it other than to not change it until after this PR, which kind of defeats the purpose a bit IMO.

**Comment by Qix-** [comment]
Btw the tests _are_, indeed, passing. Travis and Coveralls have been having mini-meltdowns today.

**Comment by Qix-** [comment]
Maybe @SamVerschueren has an opinion, too.

**Comment by sindresorhus** [comment]
:+1: Looks good.

**Comment by SamVerschueren** [comment]
Like it :+1:

File: index.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+		return [h, w, b, a];
+	}
+};
+
+cs.to.hex = function (rgb) {
+	return '#' + hexDouble(rgb[0]) + hexDouble(rgb[1]) + hexDouble(rgb[2]);
+};
+
+cs.to.rgb = function () {
+	var rgba = swizzle(arguments);
```
Comment (SamVerschueren) [inline index.js]: You could use `swizzle.wrap` here

```js
cs.to.rgb = swizzle.wrap(function (rgba) {
	return rgba.length < 4 || rgba[3] === 1
		? 'rgb(' + rgba[0] + ', ' + rgba[1] + ', ' + rgba[2] + ')'
		: 'rgba(' + rgba[0] + ', ' + rgba[1] + ', ' + rgba[2] + ', ' + rgba[3] + ')';
});
```

**Comment by ben-eb** [comment]
:shipit: :+1:

File: index.js (hunk 1 above)
Comment (Qix-) [inline index.js]: I could, but to me that looks slightly more complex than it needs to be. With the explicit call to `swizzle()`, too, it's clear what's happening and that the arguments are being reconstructed in some way that would force a maintainer to see what `swizzle()` does.

**Comment by Qix-** [comment]
Published as `1.0.0`. Thanks everyone!

**Pull request merged**